### PR TITLE
fix(routing): regression on partial dynamic routes

### PR DESCRIPTION
**Pull request by ematipico**
## Changes

Closes https://github.com/withastro/astro/issues/10173

Closes PLT-1772

## Testing

I added a new test case

## Docs

N/A

### CODE REVIEW

File: packages/astro/test/units/routing/manifest.test.js [hunk 1]
```diff
@@ -143,6 +144,8 @@ describe('routing - createRouteManifest', () => {
 
 		assertRouteRelations(getManifestRoutes(manifest), [
 			['/', '/[...rest]'],
+			['/static', '/static-'],
+			['/static-', '/[dynamic]'],
```
[the remark this answers is not in the document]
Comment (ematipico) [reply packages/astro/test/units/routing/manifest.test.js:148]: yeah it's removed 😅

**Review by matthewp — APPROVED** [review]

File: .changeset/eight-spoons-nail.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+"astro": patch
+---
+
+Fixes a regression where full dynamic routes where prioritized over partial dynamic routes. Now a route like `food-[name].astro` is matched **before** `[name].astro`.
```
Comment (bluwy) [inline .changeset/eight-spoons-nail.md:5]:
```suggestion
Fixes a regression where full dynamic routes were prioritized over partial dynamic routes. Now a route like `food-[name].astro` is matched **before** `[name].astro`.
```

**Review by bluwy — APPROVED** [review]

File: packages/astro/test/units/routing/manifest.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Fryuni) [reply packages/astro/test/units/routing/manifest.test.js:148]: Sure, when I sent the tests were still running so I expected it to break. But yeah, separate PR to fix this

[the remark this answers is not in the document]
Comment (Fryuni) [reply packages/astro/test/units/routing/manifest.test.js:148]: There is already an open issue for this: https://github.com/withastro/astro/issues/10122

**Pull request merged**